flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -108,6 +111,11 @@
 
 	private final Options ALL_OPTIONS;
 
+	private static final String fileName = "yarn-app.ini";
+	private static final String cookieKey = "secureCookie";
```

Comment:
Yes, I will make the change. 
- Do you object to retain the ini file format and port the current properties file implementation to INI format (to persist multiple application states)?

- Per current implementation (retrieveCluster), the CLI code fetches the application ID from properties file if not supplied through CLI argument. When we support multiple application state, then we expect application ID to be supplied always since there could be more than one application ID and the default functionality will go away. Do you concur? 

>
If we really need to provide backward compatibility support, then we could return the application ID from the INI file should there be just only instance persisted? If more than one application ID exists, then we throw an error indicating "Application ID" needs to be supplied as CLI argument.

Please let me know how you want me to approach and I will make the changes accordingly.